eNMS/server.py

```diff
@@ -191,6 +197,20 @@ def unauthorized():
     def configure_routes(self):
         blueprint = Blueprint("blueprint", __name__, template_folder="../templates")
 
+        def redirect_after_login(default=None):
+            def is_safe_url(target):
+                endpoint = f"/{target.split('/')[1]}"
+                endpoint_rbac = app.rbac["get_requests"].get(endpoint)
+                return (
+                    endpoint_rbac
+                    and current_user.is_admin
+                    or endpoint_rbac == "access"
+                    and endpoint in getattr(current_user, "get_requests")
+                )
+
+            next_url = (request.args.get("next") or "").strip()
+            return redirect(next_url if next_url and is_safe_url(next_url) else default)
+
```

Comment:
I don't understand this :
- it is duplicated code: it is reimplementing part of the monitor requests decorator. However, if you're redirecting to a page, you should go through the monitor requests for that page anyway, so eNMS will run this duplicated code twice ?
- It has an unwanted side-effect: if you request a page that does not exist, you'll get automatically redirected to the "default" URL. But if you were already logged in, eNMS wouldn't run this code, so you would get a 404 error instead. So for the same URL, the app renders a different page depending on whether you're already logged in, or getting redirected from the login page... This is inconsistent.
- It takes a keyword argument set to `None`, but the function is only used once with that keyword argument set to `dashboard`, so why is this an argument ?